coordinator/shard_mapper.go

```diff
@@ -0,0 +1,181 @@
+package coordinator
+
+import (
+	"io"
+	"time"
+
+	"github.com/influxdata/influxdb/influxql"
+	"github.com/influxdata/influxdb/services/meta"
+	"github.com/influxdata/influxdb/tsdb"
+)
+
+// IteratorCreator is an interface that combines mapping fields and creating iterators.
+type IteratorCreator interface {
+	influxql.IteratorCreator
+	influxql.FieldMapper
+	io.Closer
+}
+
+// ShardMapper retrieves and maps shards into an IteratorCreator that can later be
+// used for executing queries.
+type ShardMapper interface {
+	MapShards(sources influxql.Sources, opt *influxql.SelectOptions) (IteratorCreator, error)
+}
```

Comment:
Why is `ShardMapper` necessary? Does the query engine's interface need to know about shards?